### PR TITLE
Marathon auth + self signed certs + junction new version.

### PR DESCRIPTION
**Pull request by ankanm**
* Ability to turn on basic auth in marathon (Issue #2) as well as turn on SSL.
* Added an ansible play bool to generate self signed certs (not intended for production usage.)
* Switched to newer bamboo version that has HTTP_PREFIX.

### CODE REVIEW

File: ansible/roles/cert-generator/templates/gencert.sh.j2 [hunk 1]
```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+# Adapted/modified (to work with ansible and create Java keystore) on 2015-12-28
+# by Ankan Mukherjee from the original here - https://gist.github.com/bradland/1690807
+##
+# WARNING:
+#  This script was created only for dev/test purposes. It *DELETES* existing keys and certs.
```
Comment (arunabha) [inline ansible/roles/cert-generator/templates/gencert.sh.j2]: Do we have a way of using a provided cert ? Using a self signed cert will result in browser warnings.

Comment (ankanm) [inline ansible/roles/cert-generator/templates/gencert.sh.j2]: Yes. Absoluely. The regular marathon install playbook now asks for an ssl keystore location & keystore password and then updates configuration (it doesn't matter how the cert in the keystore was created.)
Typically the cert would be generated via a regular route of keygen -> csr -> trusted root ca etc. This self signed cert generation playbook is purely for dev/test reasons (and optional.) And yes, the browser will warn with this. So, not intended for prod usage anyway.

**Comment by arunabha** [comment]
Lgtm

**Comment by rvenkatesh25** [comment]
Looks good.

For SSL termination: If we are putting a proxy in front of framework UI for authn purposes, it will be a good idea to use the same proxy for the SSL termination as well (w/ a combined SAN based fat certificate), that way we don't have to get into the specifics of how each framework handles SSL. 

Are the Marathon and Chronos UI also put behind HAProxy? If so we can use:
```bash
HAProxy (SSL termination) -> oauth2_proxy -> UI
```

else, if we are setting up a new one (and not worried about dynamic discovery for frameworks):
```bash
Nginx (SSL termination) -> oauth2_proxy -> UI
```

In any case, this PR is a required intermediate step.

**Comment by ankanm** [comment]
> Are the Marathon and Chronos UI also put behind HAProxy?

They are not but they can be. I think we ought to do that anyway.

**Pull request merged**